src/Monolog/Handler/BrowserConsoleHandler.php

```diff
@@ -167,18 +167,19 @@ private static function generateScript(): string
 
     private static function handleStyles(string $formatted): array
     {
-        $args = [static::quote('font-weight: normal')];
+        $args = [];
         $format = '%c' . $formatted;
         preg_match_all('/\[\[(.*?)\]\]\{([^}]*)\}/s', $format, $matches, PREG_OFFSET_CAPTURE | PREG_SET_ORDER);
 
         foreach (array_reverse($matches) as $match) {
-            $args[] = static::quote(static::handleCustomStyles($match[2][0], $match[1][0]));
-            $args[] = '"font-weight: normal"';
+            array_unshift($args, '"font-weight: normal"');
```

Comment:
I don't see why you use `array_unshift` instead of `$arr[] = ...`

If you weren't use reversed styles maybe then would be better to "unreverse" it back? Because recalculating array index (**O(n)**) on every iteration looks terrible...

Reply:
The reason for doing it in reverse is to be able to substr in the %c placeholders a few lines below which I've not touched.

I guess the easiest alternative would be to build up args, then reverse as the last action.

Reply:
I guess it would be better.

P.S. Did you understand why `array_reverse($matches)` is used? Because I don't :(

Reply:
Yeah, it's because of the way the string substitution for replacing the format specifier matches with the console %c placeholder works based on the offsets returned by preg_match_all. In order for the offsets to remain correct as each match is processed it has to work from the end of the string. It's possible that this could be achieved in some other manner but I didn't fancy touching it!

I've pushed a change to replace the array_unshifts with a single array_reverse on return which is indeed a bit nicer.